code/game/objects/items.dm

```diff
@@ -708,7 +708,9 @@
 	action.Grant(to_who)
 
 /// Sometimes we only want to grant the item's action if it's equipped in a specific slot.
-/obj/item/proc/item_action_slot_check(slot, mob/user)
+/obj/item/proc/item_action_slot_check(slot, datum/action/action, mob/user)
+	if(SEND_SIGNAL(src, COMSIG_ITEM_UI_ACTION_SLOT_CHECKED, user, action, slot) & COMPONENT_ITEM_ACTION_SLOT_INVALID)
+		return FALSE
```

Comment:
This kinda kills me, as this proc does not enforce calling parent (and it shouldn't).

Maybe it should be in `give_item_action` instead? Change it into `COMSIG_ITEM_ACTION_GIVEN_OUT` and `return COMPONENT_BLOCK_ACTION_HANDOUT` etc.